engine/dbscan/src/main/java/org/opennms/alec/engine/dbscan/DBScanEngineFactory.java

```diff
@@ -46,7 +68,22 @@ public String getName() {
 
     @Override
     public AbstractClusterEngine createEngine(MetricRegistry metrics) {
-        return new DBScanEngine(metrics, epsilon, alpha, beta);
+        if(bundleContext != null) {
+            try {
+                ServiceReference<?>[] distanceMeasureRefs = bundleContext.getAllServiceReferences(DistanceMeasureFactory.class.getCanonicalName(), null);
+                for (ServiceReference<?> distanceMeasureRef : distanceMeasureRefs) {
+                    DistanceMeasureFactory factory = (DistanceMeasureFactory) bundleContext.getService(distanceMeasureRef);
+                    if (factory.getName().equalsIgnoreCase(distanceMeasureFactory)) {
+                        return new DBScanEngine(metrics, epsilon, alpha, beta, factory);
+                    }
+                }
+            } catch (InvalidSyntaxException e) {
+                throw new RuntimeException(e);
+            }
+
+            LOG.error("Wrong distance measure configuration {}, we'll use default {}", distanceMeasureFactory, AlarmInSpaceAndTimeDistanceMeasureFactory.class.getCanonicalName());
+        }
+        return new DBScanEngine(metrics, epsilon, alpha, beta, new AlarmInSpaceAndTimeDistanceMeasureFactory());
```

Comment:
Shouldn't we throw exception if we don't have any ServiceReference for the factories?

if `AlarmInSpaceAndTimeDistanceMeasureFactory()` is the default one to use, make it member variable.